pkg/cmd/listen.go

```diff
@@ -43,9 +43,10 @@ type listenCmd struct {
 	skipUpdate            bool
 	apiBaseURL            string
 	noWSS                 bool
+	parentContext         context.Context
```

Comment:
[Storing contexts in a struct is generally discouraged](https://github.com/golang/go/wiki/CodeReviewComments#contexts), because it makes it hard to reason about the state of the system or lifetimes of in-flight work. One of the exceptions is if it's necessary to meet an interface you don't control, which would apply here, but I _think_ we should be able to solve it without this.

Reply:
Yes, I agree & I had seen that and thought that I must be reasoning about this incorrectly

so I spoke with @pepin-stripe just now & was informed that the original playback code was written before #574 which is why we had to work around the Cobra CLI interface rather than with it (running the listen command from within the playback command shouldn't be happening with Cobra)

So next step is to undo all of his PR and call the core proxy logic from the playback command rather than the listen command